File: TextosJapones/src/main/java/org/jisho/textosJapones/model/services/KanjiServices.java
package org.jisho.textosJapones.model.services;

import org.jisho.textosJapones.database.dao.DaoFactory;
import org.jisho.textosJapones.database.dao.KanjiDao;
import org.jisho.textosJapones.model.entities.Kanji;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

public class KanjiServices {

    private final KanjiDao kanjiDao = DaoFactory.createKanjiDao();

    public Kanji select(String kanji) throws ExcessaoBd {
        return kanjiDao.select(kanji);
    }

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/service/KanjiServices.kt
package br.com.fenix.processatexto.service

import br.com.fenix.processatexto.database.DaoFactory
import br.com.fenix.processatexto.database.dao.KanjiDao
import br.com.fenix.processatexto.model.entities.processatexto.Kanji
import java.sql.SQLException
import java.util.*


class KanjiServices {

    private val kanjiDao: KanjiDao = DaoFactory.createKanjiDao()

    @Throws(SQLException::class)
    fun select(kanji: String): Optional<Kanji> = kanjiDao.select(kanji)

}